test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -506,29 +507,16 @@ func (rc *ResourceConsumer) WaitForReplicas(ctx context.Context, desiredReplicas
 // EnsureDesiredReplicasInRange ensure the replicas is in a desired range
 func (rc *ResourceConsumer) EnsureDesiredReplicasInRange(ctx context.Context, minDesiredReplicas, maxDesiredReplicas int, duration time.Duration, hpaName string) {
 	interval := 10 * time.Second
-	err := wait.PollUntilContextTimeout(ctx, interval, duration, true, func(ctx context.Context) (bool, error) {
-		replicas := rc.GetReplicas(ctx)
-		framework.Logf("expecting there to be in [%d, %d] replicas (are: %d)", minDesiredReplicas, maxDesiredReplicas, replicas)
-		as, err := rc.GetHpa(ctx, hpaName)
-		if err != nil {
-			framework.Logf("Error getting HPA: %s", err)
-		} else {
-			framework.Logf("HPA status: %+v", as.Status)
-		}
-		if replicas < minDesiredReplicas {
-			return false, fmt.Errorf("number of replicas below target")
-		} else if replicas > maxDesiredReplicas {
-			return false, fmt.Errorf("number of replicas above target")
-		} else {
-			return false, nil // Expected number of replicas found. Continue polling until timeout.
-		}
-	})
-	// The call above always returns an error, but if it is timeout, it's OK (condition satisfied all the time).
-	if wait.Interrupted(err) {
-		framework.Logf("Number of replicas was stable over %v", duration)
-		return
+	framework.Gomega().Consistently(ctx, func(ctx context.Context) int {
+		return rc.GetReplicas(ctx)
+	}).WithTimeout(duration).WithPolling(interval).Should(gomega.And(gomega.BeNumerically(">=", minDesiredReplicas), gomega.BeNumerically("<=", maxDesiredReplicas)))
```

Comment:
This returns an error that would need to returned by `EnsureDesiredReplicasInRange`. As you are not doing that yet, capture it in `err` and check it below.

Reply:
updated